lib/extend/tag.js

```diff
@@ -77,7 +75,7 @@ class NunjucksBlock extends NunjucksTag {
   }
 
   run(context, args, body) {
-    return this._run(context, args, trimBody(body));
+    return this._run(context, args, trimBody(body)) + '\n';
```

Comment:
Why add `\n`?

Look at the following test cases, which one is expected?
```js
    const str = [
      'begin',
      '{% test foo bar %}',
      'test content',
      '{% endtest %}',
      'end'
    ].join('\n');
    const result = await tag.render(str);

    // there is only one `\n` before and after
    result.should.eql('begin\nfoo bar test content\nend');
    // Or should there be two `\n` at the end?
    result.should.eql('begin\nfoo bar test content\n\nend');
```

Reply:
This extra `\n` is to separete a Nunjucks tag from immediate-after content.
So I expect *two* `\n` at the end.

Please refer the above comment https://github.com/hexojs/hexo/pull/4171#issuecomment-595027051